File: hello-mybatis/src/main/java/com/yicj/mysql/controller/UserController.java
package com.yicj.mysql.controller;

import com.yicj.mysql.entity.UserInfo;
import com.yicj.mysql.service.UserService;
import lombok.extern.slf4j.Slf4j;
import org.reactivestreams.Subscriber;
import org.reactivestreams.Subscription;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;
import reactor.core.publisher.Flux;

import java.util.List;

@Slf4j
@RestController
public class UserController {

    @Autowired
    private UserService userService ;

    @PostMapping("/user/findList")
    public Flux<UserInfo> findList(@RequestBody UserInfo userInfo){
        log.info("===> findList method exec start ....");
        Flux<UserInfo> flux = Flux.create(emitter -> {
            List<UserInfo> list = userService.findList(userInfo);
            for (UserInfo user : list) {
                emitter.next(user);
            }
            emitter.complete();
        });

        flux.subscribe(new Subscriber<UserInfo>() {
            private Subscription subscription ;
            @Override
            public void onSubscribe(Subscription subscription) {
                log.info("onSubscribe ..");
                this.subscription = subscription ;
                this.subscription.request(1);
            }

            @Override
            public void onNext(UserInfo userInfo) {
                log.info("onNext : {}", userInfo);
                this.subscription.request(1);
            }

            @Override
            public void onError(Throwable throwable) {
                log.info("onError ..");
            }

            @Override
            public void onComplete() {
                log.info("onComplete ..");
            }
        }) ;

        log.info("===> findList method exec end ....");

        return flux ;
    }
}

File: hello-kotlin/src/main/java/com/yicj/kotlin/controller/UserController.kt
package com.yicj.kotlin.controller

import com.yicj.kotlin.model.UserDO
import com.yicj.kotlin.service.UserService
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/user")
class UserController {
    var log: Logger = LoggerFactory.getLogger(UserController::class.java) ;

    @Autowired
    lateinit var userService: UserService ;

    @GetMapping("/hello")
    fun hello(): String{
        var userDO = UserDO(name = "张三", age = 18)
        //userDO.name = "张三"
        //userDO.age = 20
        log.info("user do : {}", userDO)
        var name = "yicj"
        userService.saveUser(name)
        return "hello" ;
    }
}